pkg/execution/driver/httpdriver/dialer.go

```diff
@@ -0,0 +1,128 @@
+package httpdriver
+
+import (
+	"context"
+	"fmt"
+	"net"
+)
+
+var privateIPBlocks []*net.IPNet
+var nat64blocks []*net.IPNet
+
+func init() {
+	for _, cidr := range []string{
+		"127.0.0.0/8",    // IPv4 loopback
+		"10.0.0.0/8",     // RFC1918
+		"172.16.0.0/12",  // RFC1918
+		"192.168.0.0/16", // RFC1918
+		"169.254.0.0/16", // RFC3927 link-local
+		"::1/128",        // IPv6 loopback
+		"fe80::/10",      // IPv6 link-local
+		"fc00::/7",       // IPv6 unique local addr
+		"ff00::/8",       // multicast
```

Comment:
Why not use [`IsMulticast`](https://pkg.go.dev/net#IP.IsMulticast)?

Reply:
we can, but this is now one loop for all banned IPs.